### ISSUE TITLE
Add kDrive (Infomaniak) storage type

### ISSUE THREAD

**Issue by yannicka** [hwittenborn/celeste#39]
I use [kDrive](https://www.infomaniak.com/en/kdrive) as cloud provider. It would be nice to be able to use Celeste for synchronisation.

**Comment by hwittenborn** [comment]
It doesn't look like the backend library Celeste uses to sync files currently supports kDrive. It looks like [kDrive supports WebDAV though](https://www.infomaniak.com/en/support/faq/2409/connect-to-kdrive-via-webdav), so you should be able to use that. Would that work?

**Comment by gregorydk** [comment]
@hwittenborn That would not work. You can connect to kDrive via WebDAV but nothing shows up.

![image](https://user-images.githubusercontent.com/7804464/218098209-1b4c1a5a-de85-43a2-96e2-92e0e04270bf.png)

**Comment by hwittenborn** [comment]
> @hwittenborn That would not work. You can connect to kDrive via WebDAV but nothing shows up.

Have you tried clicking the folder add button?

**Comment by gregorydk** [comment]
> > @hwittenborn That would not work. You can connect to kDrive via WebDAV but nothing shows up.
> 
> Have you tried clicking the folder add button?

Yes, Celeste freezes and becomes unresponsive (completely grayed out) when clicking the folder add button.

**Comment by gregorydk** [comment]
Syncing works if you add an existing local folder to an already existing remote folder.

Adding another folder freezes and makes the app unresponsive, the dialogue to add a folder doesn't show up.

The added folder is in an 'Syncing / Finished sync checks' loop animation. Is that normal?

**Comment by hwittenborn** [comment]
> Syncing works if you add an existing local folder to an already existing remote folder.

Celeste requires any folders to exist on both the local and remote systems - there's plans to add support for automatically creating the remote folder when it doesn't exist, but that hasn't happened yet (though it should be in the near future).

> Adding another folder freezes and makes the app unresponsive, the dialogue to add a folder doesn't show up.

It might take a second depending on your internet speed - Celeste scans the remote to check what folders are available before showing the window, so that it can offer suggestions on what remote folder to sync to. How long have you tried waiting?

> The added folder is in an 'Syncing / Finished sync checks' loop animation. Is that normal?

If it's where I think you're talking about then yeah, it's just reporting when syncing starts and when it finishes. Could you send a screenshot of the area you're talking about just so I can make sure though?

**Comment by gregorydk** [comment]
After doing some more testing...

It appears to work but adding the drive or a directory takes a long time, up to a minute. The syncing is instantaneously.

The icon in Celeste next to the directory is in an infinite spinning / check mark loop.

**Comment by hwittenborn** [comment]
> The icon in Celeste next to the directory is in an infinite spinning / check mark loop.

That's just an issue with Celeste's current syncing implementation: after all directories are synced, it just starts from the beginning and checks for more changes. It needs to be moved over to only syncing once files have been changed on the local/remote system, but that hasn't been implemented yet.

> It appears to work but adding the drive or a directory takes a long time, up to a minute.

Would you be able to start up Celeste from a terminal and see if anything pops up during the minute after you click the add directory button?

**Comment by gregorydk** [comment]
Started Celeste (flatpak) from a terminal, no extra-ordinary output. The Remote Folder Picker popped up after 4 minutes!

    (celeste:23): Adwaita-CRITICAL **: 18:12:50.624: ../src/adw-header-bar.c:640: tried to remove non-child 0x55b5d654f300 of type 'GtkButton' from 0x55b5d6515340 of type 'AdwHeaderBar'
    Info: Starting up tray binary...
